app/state.go

```diff
@@ -1104,6 +1109,68 @@ func (fir *fetchStoreRetryState) Handle(ctx *StateContext, c Controller) (State,
 	return fir.Wait(NewUpdateFetchState(&fir.update), fir, intvl, ctx.WakeupChan)
 }
 
+type inventoryUpdateRetry struct {
+	baseState
+	WaitState
+	from State
+	err  error
+}
+
+func NewInventoryUpdateRetryState(from State, update *datastore.UpdateInfo,
+	err error) State {
+	return &inventoryUpdateRetry{
+		baseState: baseState{
+			id: datastore.MenderStateInventoryUpdateRetryWait,
+			t:  ToSync,
+		},
+		WaitState: NewWaitState(datastore.MenderStateInventoryUpdateRetryWait, ToSync),
+		from:      from,
+		err:       err,
+	}
+}
+
+func (fir *inventoryUpdateRetry) Cancel() bool {
+	return fir.WaitState.Cancel()
+}
+func (fir *inventoryUpdateRetry) Handle(ctx *StateContext, c Controller) (State, bool) {
+	if ctx.nextAttemptAt.After(time.Now()) {
+		secondsRemaining := time.Duration(ctx.nextAttemptAt.Unix()-time.Now().Unix()) * time.Second
```

Comment:
Should use: `secondsRemaining := ctx.nextAttemptAt.Sub(time.Now()) * time.Second`

Reply:
corrected. (dropped the time.Second() as Sub returns duration)

Reply:
and the static checker was smarter -- suggested Until